hashes/src/macros.rs

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Public macros.
+
+/// Adds trait impls to a bytelike type.
+///
+/// Implements:
+///
+/// * `str::FromStr`
+/// * `fmt::{LowerHex, UpperHex}` using `hex-conservative`.
+/// * `fmt::{Display, Debug}` by calling `LowerHex`
```

Comment:
Should be gated on `hex-conservative` feature.

Reply:
Feature gating won't work, as far as I understand, if the macro is in a different crate to the caller and the caller does not have that feature eg `bitcoin` does not have a `hex` feature. Also, in its current state, this macro is no good for crates that do not use `hex` because the `Debug` impl depends on hex stuff.